platform-sdk/hedera-nativesupport/src/main/java/com/hedera/nativesupport/LibraryLoader.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (C) 2023-2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.nativesupport;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Utility class that helps to extract a library contained in the JAR file into a temporary directory.
+ */
+public final class LibraryLoader {
+
+    /**
+     * The root resources folder where the software is located.
+     */
+    private static final String SOFTWARE_FOLDER_NAME = "software";
+
+    /**
+     * The path delimiter used in the JAR file.
+     */
+    private static final String RESOURCE_PATH_DELIMITER = File.separator;
+
+    /**
+     * The {@link ResourceLoader} used to load the Helm executable.
+     */
+    private static final ResourceLoader RESOURCE_LOADER = new ResourceLoader(LibraryLoader.class);
```

Comment:
This assumes the resources are always located in the jar file containing the `LibraryLoader` class.